legacy/firmware/config.c

```diff
@@ -772,12 +784,69 @@ bool config_changeWipeCode(const char *pin, const char *wipe_code) {
   return sectrue == ret;
 }
 
-const uint8_t *session_getSessionId(void) {
-  if (!sessionIdCached) {
-    random_buffer(sessionId, 32);
+uint8_t session_findLRU(void) {
+  uint8_t lowest_index = MAX_SESSIONS_COUNT;
+  uint32_t lowest = sessionLRUCounter;
+  for (uint8_t i = 0; i < MAX_SESSIONS_COUNT; i++) {
+    if (sessionsCache[i].usage < lowest) {
+      lowest = sessionsCache[i].usage;
+      lowest_index = i;
+    }
+  }
```

Comment:
I find this really hard to read. Rename:
lowest -> oldest_use
lowest_index -> oldest_index
.usage -> .last_use

Reply:
I agree with the first two. About the third one I am not sure. How about `usage_counter`? Or `use_count`? `last_use` feels like the highest number is the last used, but it is the opposite.

Reply:
I am not sure I understand. I thought `.usage` is meant to record when the session was last used. It doesn't count the number of times it was used. If it did, then you would be discarding the least used session, and we agreed we don't want that.

BTW, I am open to any equivalent alternatives for the first two, e.g. `least_recent_use` and `least_recent_index`, as long as the name says something about the variable. I can't think of any alternatives for the third though.

Reply:
Ok, I believe those are the best:

lowest -> least_recent_use
lowest_index -> least_recent_index
.usage -> .last_use

Reply:
It would be nice to rename `sessionLRUCounter` as well to fit this new names. `sessionUseCounter`?

Reply:
I totally agree.